primitives/core/src/traits.rs

```diff
@@ -69,11 +82,70 @@ pub trait BareCryptoStore: Send + Sync {
 
 	/// Get the password for this store.
 	fn password(&self) -> Option<&str>;
-
+	/// Find intersection between provided keys and supported keys
+	///
+	/// Provided a list of (CryptoTypeId,[u8]) pairs, this would return
+	/// a filtered set of public keys which are supported by the keystore.
+	fn supported_keys(&self, id: KeyTypeId, keys: Vec<CryptoTypePublicPair>) -> Result<HashSet<CryptoTypePublicPair>, Error>;
+	/// List all supported keys
+	///
+	/// Returns a set of public keys the signer supports.
+	fn keys(&self, id: KeyTypeId) -> Result<HashSet<CryptoTypePublicPair>, Error>;
 	/// Checks if the private keys for the given public key and key type combinations exist.
 	///
 	/// Returns `true` iff all private keys could be found.
 	fn has_keys(&self, public_keys: &[(Vec<u8>, KeyTypeId)]) -> bool;
+
+	/// Sign with key
+	///
+	/// Signs a message with the private key that matches
+	/// the public key passed.
+	fn sign_with(
```

Comment:
It should be mentioned what this function returns and I think it would be easier to return the SCALE encoded signature (instead of the raw bytes).

Reply:
I had a look at how we use the codec crate... `Encode::encode` already returns a `Vec<u8>` as well. I am pretty sure i don't follow what you're suggesting. Is there code somewhere i can look into that does this?